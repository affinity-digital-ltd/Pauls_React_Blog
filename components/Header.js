import { Component } from 'react'
import Link from 'next/link'

import HeaderNav from './HeaderNav'

class Header extends Component {
  render () {
    return <header className='c-header'>
      <div className='o-header__container'>
        <Link href='/index'><a><img src='static/images/logo.png' alt='Paul Brighton Web Development Logo' className='c-brand-logo' /></a></Link>
        <div className='c-subscribe-form'>
          <form className='c-subscribe-form__container'>
            <input type='email' value='Email' placeholder='Email' name='EMAIL' className='c-subscribe-form__input' id='mce-EMAIL' />
            <input type='submit' value='Subscribe' name='subscribe' className='c-subscribe-form__btn' />
          </form>
        </div>
      </div>
      <HeaderNav />
    </header>
  }
}

export default Header
