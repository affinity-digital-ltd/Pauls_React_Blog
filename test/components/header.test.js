import Header from '../../components/Footer'

describe('<Header />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <Header />
    )

    expect(component).toMatchSnapshot()
  })
})
