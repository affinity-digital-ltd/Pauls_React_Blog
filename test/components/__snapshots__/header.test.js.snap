// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="o-header__container"
        >
          <Link
            href="/"
          >
            <a>
              <img
                alt="Paul Brighton Web Development Logo"
                className="c-brand-logo"
                src="/static/images/logo.png"
              />
            </a>
          </Link>
          <div
            className="c-subscribe-form"
          >
            <form
              action="https://twitter.us17.list-manage.com/subscribe/post?u=5e14f42c2626123e850bb543f&id=5dd22b22a9"
              className="c-subscribe-form__container"
              method="post"
              name="mc-embedded-subscribe-form"
              noValidate={true}
              target="_blank"
            >
              <input
                className="c-subscribe-form__input"
                id="mce-EMAIL"
                name="EMAIL"
                placeholder="Email"
                type="email"
              />
              <input
                className="c-subscribe-form__btn"
                name="subscribe"
                type="submit"
                value="Subscribe"
              />
            </form>
          </div>
        </div>,
        <HeaderNav />,
      ],
      "className": "c-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              href="/"
            >
              <a>
                <img
                  alt="Paul Brighton Web Development Logo"
                  className="c-brand-logo"
                  src="/static/images/logo.png"
                />
              </a>
            </Link>,
            <div
              className="c-subscribe-form"
            >
              <form
                action="https://twitter.us17.list-manage.com/subscribe/post?u=5e14f42c2626123e850bb543f&id=5dd22b22a9"
                className="c-subscribe-form__container"
                method="post"
                name="mc-embedded-subscribe-form"
                noValidate={true}
                target="_blank"
              >
                <input
                  className="c-subscribe-form__input"
                  id="mce-EMAIL"
                  name="EMAIL"
                  placeholder="Email"
                  type="email"
                />
                <input
                  className="c-subscribe-form__btn"
                  name="subscribe"
                  type="submit"
                  value="Subscribe"
                />
              </form>
            </div>,
          ],
          "className": "o-header__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a>
                <img
                  alt="Paul Brighton Web Development Logo"
                  className="c-brand-logo"
                  src="/static/images/logo.png"
                />
              </a>,
              "href": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="Paul Brighton Web Development Logo"
                  className="c-brand-logo"
                  src="/static/images/logo.png"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Paul Brighton Web Development Logo",
                  "className": "c-brand-logo",
                  "src": "/static/images/logo.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form
                action="https://twitter.us17.list-manage.com/subscribe/post?u=5e14f42c2626123e850bb543f&id=5dd22b22a9"
                className="c-subscribe-form__container"
                method="post"
                name="mc-embedded-subscribe-form"
                noValidate={true}
                target="_blank"
              >
                <input
                  className="c-subscribe-form__input"
                  id="mce-EMAIL"
                  name="EMAIL"
                  placeholder="Email"
                  type="email"
                />
                <input
                  className="c-subscribe-form__btn"
                  name="subscribe"
                  type="submit"
                  value="Subscribe"
                />
              </form>,
              "className": "c-subscribe-form",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "action": "https://twitter.us17.list-manage.com/subscribe/post?u=5e14f42c2626123e850bb543f&id=5dd22b22a9",
                "children": Array [
                  <input
                    className="c-subscribe-form__input"
                    id="mce-EMAIL"
                    name="EMAIL"
                    placeholder="Email"
                    type="email"
                  />,
                  <input
                    className="c-subscribe-form__btn"
                    name="subscribe"
                    type="submit"
                    value="Subscribe"
                  />,
                ],
                "className": "c-subscribe-form__container",
                "method": "post",
                "name": "mc-embedded-subscribe-form",
                "noValidate": true,
                "target": "_blank",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "c-subscribe-form__input",
                    "id": "mce-EMAIL",
                    "name": "EMAIL",
                    "placeholder": "Email",
                    "type": "email",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "c-subscribe-form__btn",
                    "name": "subscribe",
                    "type": "submit",
                    "value": "Subscribe",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "form",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="o-header__container"
          >
            <Link
              href="/"
            >
              <a>
                <img
                  alt="Paul Brighton Web Development Logo"
                  className="c-brand-logo"
                  src="/static/images/logo.png"
                />
              </a>
            </Link>
            <div
              className="c-subscribe-form"
            >
              <form
                action="https://twitter.us17.list-manage.com/subscribe/post?u=5e14f42c2626123e850bb543f&id=5dd22b22a9"
                className="c-subscribe-form__container"
                method="post"
                name="mc-embedded-subscribe-form"
                noValidate={true}
                target="_blank"
              >
                <input
                  className="c-subscribe-form__input"
                  id="mce-EMAIL"
                  name="EMAIL"
                  placeholder="Email"
                  type="email"
                />
                <input
                  className="c-subscribe-form__btn"
                  name="subscribe"
                  type="submit"
                  value="Subscribe"
                />
              </form>
            </div>
          </div>,
          <HeaderNav />,
        ],
        "className": "c-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                href="/"
              >
                <a>
                  <img
                    alt="Paul Brighton Web Development Logo"
                    className="c-brand-logo"
                    src="/static/images/logo.png"
                  />
                </a>
              </Link>,
              <div
                className="c-subscribe-form"
              >
                <form
                  action="https://twitter.us17.list-manage.com/subscribe/post?u=5e14f42c2626123e850bb543f&id=5dd22b22a9"
                  className="c-subscribe-form__container"
                  method="post"
                  name="mc-embedded-subscribe-form"
                  noValidate={true}
                  target="_blank"
                >
                  <input
                    className="c-subscribe-form__input"
                    id="mce-EMAIL"
                    name="EMAIL"
                    placeholder="Email"
                    type="email"
                  />
                  <input
                    className="c-subscribe-form__btn"
                    name="subscribe"
                    type="submit"
                    value="Subscribe"
                  />
                </form>
              </div>,
            ],
            "className": "o-header__container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a>
                  <img
                    alt="Paul Brighton Web Development Logo"
                    className="c-brand-logo"
                    src="/static/images/logo.png"
                  />
                </a>,
                "href": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt="Paul Brighton Web Development Logo"
                    className="c-brand-logo"
                    src="/static/images/logo.png"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "Paul Brighton Web Development Logo",
                    "className": "c-brand-logo",
                    "src": "/static/images/logo.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <form
                  action="https://twitter.us17.list-manage.com/subscribe/post?u=5e14f42c2626123e850bb543f&id=5dd22b22a9"
                  className="c-subscribe-form__container"
                  method="post"
                  name="mc-embedded-subscribe-form"
                  noValidate={true}
                  target="_blank"
                >
                  <input
                    className="c-subscribe-form__input"
                    id="mce-EMAIL"
                    name="EMAIL"
                    placeholder="Email"
                    type="email"
                  />
                  <input
                    className="c-subscribe-form__btn"
                    name="subscribe"
                    type="submit"
                    value="Subscribe"
                  />
                </form>,
                "className": "c-subscribe-form",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "action": "https://twitter.us17.list-manage.com/subscribe/post?u=5e14f42c2626123e850bb543f&id=5dd22b22a9",
                  "children": Array [
                    <input
                      className="c-subscribe-form__input"
                      id="mce-EMAIL"
                      name="EMAIL"
                      placeholder="Email"
                      type="email"
                    />,
                    <input
                      className="c-subscribe-form__btn"
                      name="subscribe"
                      type="submit"
                      value="Subscribe"
                    />,
                  ],
                  "className": "c-subscribe-form__container",
                  "method": "post",
                  "name": "mc-embedded-subscribe-form",
                  "noValidate": true,
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "c-subscribe-form__input",
                      "id": "mce-EMAIL",
                      "name": "EMAIL",
                      "placeholder": "Email",
                      "type": "email",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "c-subscribe-form__btn",
                      "name": "subscribe",
                      "type": "submit",
                      "value": "Subscribe",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "form",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
